back/app/services/user_slug_service.rb

```diff
@@ -2,7 +2,7 @@
 
 class UserSlugService
   def generate_slug(user, string)
-    return SecureRandom.uuid if abbreviated_user_names?
+    return SecureRandom.uuid if abbreviated_user_names? || string == ''
```

Comment:
Should we overwrite the UUID slug when a user upgrades from light resident to full resident?

Reply:
I'm going to come onto this in another branch I think